### PR TITLE
Add CAPZ presubmit for ipv6 ci conformance

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
Add optional job to test https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4086, will eventually convert into a periodic job to test every branch of k8s

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 1]
```diff
@@ -313,6 +313,49 @@ presubmits:
...
+      containers:
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230727-ea685f8747-1.27
+          command:
+            - runner.sh
+          args:
+            - ./scripts/ci-conformance.sh
+          env:
+            - name: E2E_ARGS
+              value: "-kubetest.use-ci-artifacts"
+            - name: IPV6
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:348]: this is the main difference between this job and the one above (pull-cluster-api-provider-azure-conformance-with-ci-artifacts)

**Review by CecileRobertMichon — COMMENTED** [review]
/assign @mboersma @willie-yao @nawazkh

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:348]: /hold 

going to change this

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml]: /hold cancel

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 2]
```diff
@@ -313,6 +313,49 @@ presubmits:
...
+      containers:
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230727-ea685f8747-1.27
+          command:
+            - runner.sh
+          args:
+            - ./scripts/ci-conformance.sh
+          env:
+            - name: E2E_ARGS
+              value: "-kubetest.use-ci-artifacts"
+            - name: IP_FAMILY
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:348]: this is the main difference between this job and the one above (pull-cluster-api-provider-azure-conformance-with-ci-artifacts)

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 3]
```diff
@@ -313,6 +313,49 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: capz-pr-conformance-k8s-ci-main
+  - name: pull-cluster-api-provider-azure-conformance-ipv6-with-ci-artifacts
+    path_alias: "sigs.k8s.io/cluster-api-provider-azure"
+    always_run: false
+    optional: true
+    decorate: true
+    run_if_changed: 'test\/e2e|templates\/test|scripts\/ci-conformance.sh|Makefile'
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:321]: /hold

I need to remove this

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml]: /hold cancel

**Review by willie-yao — COMMENTED** [review]
Looks good! Just one question on my end: Does this test need this line?
```
run_if_changed: 'test\/e2e|templates\/test|scripts\/ci-conformance.sh|Makefile'
```
Since the test above has it and it is also a conformance test

**Comment by CecileRobertMichon** [comment]
@willie-yao good question, I removed that line so it wouldn't get triggered on every PR that changes those files, since the test won't fully work until https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4086 merges

**Pull request merged**